contracts/treasury/ZNSTreasury.sol

```diff
@@ -156,13 +156,21 @@ contract ZNSTreasury is AAccessControlled, ARegistryWired, UUPSUpgradeable, IZNS
      * Since we are clearing storage, gas refund from this operation makes Revoke transactions cheaper.
      * @param domainHash The hash of the domain for which the stake is being withdrawn.
      * @param owner The address of the user who is withdrawing the stake.
+     * @param protocolFee The protocol fee paid by the user to Zero.
      */
     function unstakeForDomain(
         bytes32 domainHash,
-        address owner
+        address owner,
+        uint256 protocolFee
     ) external override onlyRegistrar {
         Stake memory stakeData = stakedForDomain[domainHash];
         delete stakedForDomain[domainHash];
+        
+        stakeData.token.safeTransferFrom(
```

Comment:
what if this fee is 0?
Then we are paying for a transfer of 0 amount.
Probably needs a check.

Reply:
We only call `unstakeForDomain` if in the `_coreRevoke` flow the `stakedAmount > 0` and so will always be non-zero, unless we change the fee percentage charged in the root pricer.